File: trunk/unityClientSample/Assets/src/protocol/RoomInfo.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
	[ProtoContract]
	public class RoomInfo
	{
        /// <summary>
        ///  房间id
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public string id;
        /// <summary>
        ///  场景icon
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string background;
        /// <summary>
        ///  我方信息
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public BattleInfo attInfo;
        /// <summary>
        ///  敌方信息
        /// </summary>
		[ProtoMember(4, IsRequired = false)]
		public BattleInfo defInfo;

	}
}

File: trunk/server/src/main/java/org/ngame/protocol/domain/RoomInfo.java
/**
 * room
 */
package org.ngame.protocol.domain;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

import lombok.Data;


/**
 *
 * @author csf
 */
@Data
public class RoomInfo implements Cloneable
{

    @Protobuf(description = "房间id")
	private String id;
    @Protobuf(description = "场景icon")
    private String background;
    @Protobuf(description = "我方信息")
	private BattleInfo attInfo;
    @Protobuf(description = "敌方信息")
	private BattleInfo defInfo;
    @Protobuf(description = "随机的索引")
    private int index;
	
	public BattleInfo getBattleInfo(String pid){
		if(attInfo.getPid().equals(pid)) return attInfo;
		else if(defInfo.getPid().equals(pid)) return defInfo;
		else return null;
	}
	
	@Override
	public Object clone() {  
		try {
			 return super.clone();
		} catch (CloneNotSupportedException e) {
			e.printStackTrace();
		}
		return null;
    }  
	
	public void addIndex(int i){
		index+=i;
	}
}
